codersdk/workspacesdk/connector.go

```diff
@@ -105,17 +101,25 @@ func (tac *tailnetAPIConnector) runConnector(conn tailnetConn) {
 	tac.gracefulCtx, tac.cancelGracefulCtx = context.WithCancel(context.Background())
 	go tac.manageGracefulTimeout()
 	go func() {
-		tac.isFirst = true
 		defer close(tac.closed)
 		// Sadly retry doesn't support quartz.Clock yet so this is not
 		// influenced by the configured clock.
 		for retrier := retry.New(50*time.Millisecond, 10*time.Second); retrier.Wait(tac.ctx); {
-			tailnetClient, err := tac.dial()
+			tailnetClients, err := tac.dialer.Dial(tac.ctx, tac.tokenCtrl)
 			if err != nil {
+				if errors.Is(err, context.Canceled) {
+					continue
+				}
+				fields := []any{slog.Error(err)}
+				var sdkErr *codersdk.Error
+				if xerrors.As(err, &sdkErr) {
+					fields = append(fields, slog.Error(sdkErr))
+				}
+				tac.logger.Error(tac.ctx, "failed to dial tailnet v2+ API", fields...)
```

Comment:
I've actually not seen this pattern before does it render as expected when passed multiple errors?

Reply:
...depends on what you expect, I guess.  Logger fields are just slices internally, they don't enforce uniqueness.  It's up to the sink to decide what to do if it gets the same field key multiple times.  The human logger, I think, just writes the all the fields even if a key is repeated.